src/loaders/genome_collection/compute_genome_attribs.py

```diff
@@ -309,38 +330,70 @@ def _run_microtrait(genome_id, fna_file, batch_dir):
+          microtrait_result <- extract.traits(in_file = genome_file, out_dir = out_dir)
+          return(microtrait_result)
+        }
+        """
+        r_func = robjects.r(r_script)
+        r_result = r_func(fna_file, genome_dir)
+
+        microtrait_result = _get_r_list_element(r_result, 'microtrait_result')
+
+        trait_counts_atgranularity3 = _get_r_list_element(microtrait_result, 'trait_counts_atgranularity3')
+        trait_counts_atgranularity3_df = _r_table_to_df(trait_counts_atgranularity3)
+
+        trait_counts_atgranularity3_df.to_csv(os.path.join(genome_dir, 'trait_counts_atgranularity3.csv'), index=False)
+
+    except Exception as e:
+        raise ValueError(f'Error running microtrait on {fna_file}') from e
+    finally:
+        # remove the unpacked assembly file to save space
+        if remove_gz_file:
+            print(f'removing {fna_file}')
+            os.remove(fna_file)
+
+    selected_cols = ['microtrait_trait-value', 'microtrait_trait-displaynameshort', 'microtrait_trait-displaynamelong']
+    cell_info_df = trait_counts_atgranularity3_df[selected_cols]
+
+    # Extract the substring of the 'microtrait_trait-displaynamelong' column before the last colon character
+    # and assign it to a new 'category' column in the DataFrame
+    cell_info_df['category'] = cell_info_df['microtrait_trait-displaynamelong'].str.rsplit(':', 1).str[0]
+    # Use the microtrait_trait-displaynameshort column as the column name, microtrait_trait-displaynamelong column
+    # as the column description, and microtrait_trait-value as the cell value
+    cell_info_df = cell_info_df.rename(columns={'microtrait_trait-displaynameshort': 'col_name',
+                                                'microtrait_trait-displaynamelong': 'col_description',
+                                                'microtrait_trait-value': 'cell_val'})
+
+    categories, cells, celid, min_value, max_value = list(), list(), 0, float('inf'), float('-inf')
+    for category_name, category_df in cell_info_df.groupby('category'):
+        columns = list()
+        for i, row in category_df.iterrows():
+            columns.append(ColumnInformation(id=str(i), name=row['col_name'], description=row['col_description']))
+            cell_val = float(row['cell_val'])
+            cells.append(Cell(celid=str(celid), colid=str(i), val=cell_val))
+            min_value, max_value = min(min_value, cell_val), max(max_value, cell_val)
+            celid += 1
+        categories.append(ColumnCategory(category=category_name, columns=columns))
+
+    meta = HeatMapMeta(categories=categories, min_value=min_value, max_value=max_value)
+    row = HeatMapRow(kbase_id=genome_id, cells=cells)
+
+    with open(os.path.join(genome_dir, 'heatmap_meta.json'), "w") as heatmap_meta:
+        json.dump(dict(meta), heatmap_meta)
+
+    with open(os.path.join(genome_dir, 'heatmap_row.json'), "w") as heatmap_row:
+        json.dump(dict(row), heatmap_row)
```

Comment:
Could you share sample contents for the meta and row files?

Reply:
will do that as a comment in the next PR.

Reply:
Why not this PR? It's producing the output

Reply:
So different request - can you share a csv output example from this code?

Reply:
👍

Reply:
Sorry, I didn't mean for it to be a code comment, I just thought you'd post it here. Anyway fine as is.

Reply:
One question: what's the difference between `microtrait_trait-value` and `microtrait_trait-value1`?

Reply:
I didn't even notice `microtrait_trait-value1`. Your guess is just as good as mine.

Reply:
Should we remove it to avoid confusion / bugs?

Reply:
Actually, is that column in the original output from microtrait? Maybe when merging the tables together in pandas if there's duplicate column names one of them is suffixed with `1`?

Reply:
it comes with the original output. We should probably keep it. At least for this step, we are trying to reserve the output from the tool.

Reply:
Which table is it in? is it in the same table as `microtrait_trait-value`

Reply:
yes. they both in the `trait_counts_atgranularity3` table.

Reply:
Weird, added it to doc containing questions for @ukaraoz

Reply:
:+1: